projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -4065,27 +4065,34 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
      * @hidden @internal
      */
     public beginAddRowByIndex(rowID: any, index: number, asChild?: boolean) {
+        if (!this.rowEditable) {
+            console.warn('The grid must use row edit mode to perform row adding! Please set rowEditable to true.');
+            return;
+        }
         this.endEdit(true);
         this.cancelAddMode = false;
+        const isInPinnedArea = this.isRecordPinnedByViewIndex(index);
+        const pinIndex = this.pinnedRecords.findIndex(x => x[this.primaryKey] === rowID);
```

Comment:
Why don`t we move this into another protected method `getPinnedIndexById` as you did `getUnpinnedIndexById`

Reply:
We could but it is not necessary IMO. The function is simple enough to not require a dedicated method and I don’t think extracting it in a method would add any value. 
getUnpinnedIndexById is in a separate method  for a different reason. It needs to be overwritten in the TreeGrid since its unpinned records have a different structure that the ones in the base grid. The pinned records have the always the same structure in all grids so there’s no explicit need for a separate method.